include/nanobind/eigen/dense.h

```diff
@@ -212,12 +257,25 @@ struct type_caster<Eigen::Map<T, Options, StrideType>, enable_if_t<is_eigen_plai
                       OS = StrideType::OuterStrideAtCompileTime;
 
         int64_t inner = caster.value.stride(0),
-                outer = caster.value.stride(1);
-        (void) outer;
+                outer;
+        if constexpr (NumDimensions<T> == 1)
+            outer = caster.value.shape(0);
+        else
+            outer = caster.value.stride(1);
 
         if constexpr (T::IsRowMajor)
             std::swap(inner, outer);
 
+        if constexpr (IS == 0) {
+            assert(inner == 1);
```

Comment:
We don't include `cassert`, so I don't think it's generally valid to call that here. Do you think it's really necessary that these assertions are part of the type caster, or were these merely helpful to debug all cases on your end?